src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -157,6 +220,24 @@ public override bool TryGetPropertyValue(string name, out object value)
                 throw Error.ArgumentNull("name");
             }
 
+            if (_instanceAnnotationsPropertyInfo != null)
+            {
+                if (name == _instanceAnnotationsPropertyInfo.Name)
+                {
+                    if (_instanceAnnotationCache == null)
+                    {
+                        _instanceAnnotationCache =
+                            GetInstanceannotationContainer(_instanceAnnotationsPropertyInfo, _instance, null,create: false);
+                    }
+
+                    if(_instanceAnnotationCache != null)
+                    {
+                        value = _instanceAnnotationCache;
+                        return true;
+                    }                    
+                }
+            }
```

Comment:
This logic (and equivalent logic in TrySetPropertyValue) look wrong to me -- what is the intended semantics here?  Is this trying to return the IODataInstanceAnnotationContainer itself?